Update landing page css to match figma designs

sonalranjit opened the pull request:
- Updated new landing page design to match figma design
- Updated node dependency versions
![Screenshot 2024-10-17 at 6 20 19 PM](https://github.com/user-attachments/assets/a212f837-28bd-4d46-b13b-5cc75f970ab2)

sonalranjit committed:
Update landing page css to match figma designs

sonalranjit committed:
Fix linting error

sonalranjit committed:
Reduce heroTagline font size

sonalranjit committed:
minor css padding updates and aligning button css

tallcoleman commented on Makefile:
Great update!

tallcoleman commented on bikespace_frontend/src/components/landing/content-blocks/content-blocks.module.scss:
```diff
@@ -56,15 +59,28 @@ a.buttonLink {
   &:hover {
     border-color: $color-primary;
   }
+  padding: 1.5rem;
```
This rule on line 62 doesn't need to be added again since it's already specified above; for lines 63-64, I would move above the `&` specifiers. I think there's another one like this too – was generally trying to follow the pattern of main styles first and then conditional styles added with the `&` specifiers after.

tallcoleman commented on bikespace_frontend/src/components/landing/content-blocks/content-blocks.module.scss:
```diff
@@ -17,11 +17,14 @@
   margin-bottom: 2rem;
   max-width: $wrapper-full-width;
+  flex: 2 1 400px;
+  padding: 5%;
```
If you want more padding between the content and screen edge, I would suggest adjusting `bikespace_frontend/src/components/landing/landing-layout/landing-layout.module.scss` (line 11).

There's also a knock-on effect that with the greater padding for this component, it causes the hero button text to break into two lines on mobile:

<img src="https://github.com/user-attachments/assets/25f78777-3cd9-4c5c-b596-165ba201b25d" width=400>

sonalranjit replied:
Updated

tallcoleman commented on bikespace_frontend/src/components/landing/content-blocks/content-blocks.tsx:
For this one I would try to avoid using `<br />` elements for spacing. The changes to the flex rules have also made it not work quite right on mobile — the image gets very small and stays in 2-col format instead of responsively dropping below the buttons.

<img src="https://github.com/user-attachments/assets/25f78777-3cd9-4c5c-b596-165ba201b25d" width=400>

tallcoleman commented on bikespace_frontend/src/components/landing/footer/footer.module.scss:
This is a great improvement, especially on mobile!

tallcoleman commented on bikespace_frontend/src/components/landing/landing-layout/landing-layout.module.scss:
```diff
@@ -10,6 +10,8 @@ $even-lighter-grey: #EEE;
   margin: 0 auto;
   padding: 1rem;
   max-width: $wrapper-max-width;
+  display: flex;
```
I would try and avoid doing this and control the justification of content on a block by block basis instead of at the page level.

sonalranjit replied:
Updated

tallcoleman reviewed (changes requested):
Hi Sonal, thanks for taking a look at the design – have left some changes to review, most important is that the layout still work properly on mobile.

If you came across anything confusing when you were trying to figure out the frontend folder organization, feel free to add it to the README in the `/bikespace_frontend` folder.

xinweny commented:
Thanks and great work @sonalranjit! I think @tallcoleman has covered most of the important points.

sonalranjit committed:
Update css according to review

tallcoleman committed:
Update hero buttons to prevent container overflow

Removed width specifier since it caused the hero buttons to overflow their parent container on Mobile

tallcoleman commented on bikespace_frontend/src/components/landing/content-blocks/content-blocks.module.scss:
@sonalranjit on this one I added one small tweak - removed the `width: 125%;` rule since it was causing the buttons to overflow the parent container on mobile. Before and after:

<img width="200" alt="Screenshot 2024-10-26 at 11 01 13 AM" src="https://github.com/user-attachments/assets/1edee1e8-a6f0-455d-aeac-2c3ebe65cf1b">

<img width="200" alt="Screenshot 2024-10-26 at 11 01 48 AM" src="https://github.com/user-attachments/assets/c594db8a-e2c3-4200-84e3-9fd63f2ab2e3">

tallcoleman reviewed (approved):
Thanks @sonalranjit !
